ocs_ci/deployment/ibmcloud.py

```diff
@@ -111,3 +121,217 @@ def check_cluster_existence(self, cluster_name_prefix):
             if cl["state"] != "deleting" and cl["name"].startswith(cluster_name_prefix)
         ]
         return bool(non_term_clusters_with_prefix)
+
+
+class IBMCloudIPI(CloudDeploymentBase):
+    """
+    A class to handle IBM Cloud IPI specific deployment
+    """
+
+    DEFAULT_STORAGECLASS = "ibmc-vpc-block-10iops-tier"
+    OCPDeployment = IPIOCPDeployment
+
+    def __init__(self):
+        self.name = self.__class__.__name__
+        self.installer = None
+        super(IBMCloudIPI, self).__init__()
+        self.credentials_requests_dir = os.path.join(self.cluster_path, "creds_reqs")
+        self.pull_secret_path = os.path.join(constants.TOP_DIR, "data", "pull-secret")
+
+    def deploy_ocp(self, log_cli_level="DEBUG"):
+        """
+        Perform IBMCloudIPI OCP deployment.
+
+        Args:
+            log_cli_level (str): log level for installer (default: DEBUG)
+        """
+        self.ocp_deployment = self.OCPDeployment()
+        self.ocp_deployment.deploy_prereq()
+
+        # IBM Cloud specific prereqs
+        ibmcloud.login()
+        self.configure_cloud_credential_operator()
+        self.export_api_key()
+        self.manually_create_iam_for_vpc()
+
+        self.ocp_deployment.deploy(log_cli_level)
+        # logging the cluster UUID so that we can ask for its telemetry data
+        cluster_id = exec_cmd(
+            "oc get clusterversion version -o jsonpath='{.spec.clusterID}'"
+        )
+        logger.info(f"clusterID (UUID): {cluster_id}")
+
+    def destroy_cluster(self, log_level="DEBUG"):
+        """
+        Destroy the OCP cluster.
+
+        Args:
+            log_level (str): log level openshift-installer (default: DEBUG)
+
+        """
+        self.export_api_key()
+        logger.info("Destroying the IBM Cloud cluster")
+        super(IBMCloudIPI, self).destroy_cluster(log_level)
+        self.delete_service_id()
+        resource_group = self.get_resource_group()
+        self.delete_volumes(resource_group)
+        time.sleep(30)  # Volumes take time to fully delete
```

Comment:
@clacroix12 , can we also check if entries are actually got deleted ?

Reply:
@shylesh Updated `delete_volumes` to verify that volumes are deleted before moving on. Still need to test this, I'm deploying a cluster right now so I can do so.

Reply:
Some output from my testing:

```
23:51:21 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: ibmcloud resource groups --output json
23:51:22 - MainThread - ocs_ci.deployment.ibmcloud - INFO  - Retrieving cluster resource group
23:51:22 - MainThread - ocs_ci.deployment.ibmcloud - INFO  - Found resource group: clacroix-ibm-rv5gk
23:51:22 - MainThread - ocs_ci.deployment.ibmcloud - INFO  - Deleting volumes
23:51:22 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: ibmcloud is vols --resource-group-name clacroix-ibm-rv5gk --output json
23:51:29 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: ibmcloud is vold -f r006-7f8970e2-140c-4536-8d6a-4edcb39d3fcd r006-1a1603d0-b79e-46e5-95a4-567f182d4e70 r006-8e3fa3f1-52a3-46b4-a803-38e1b939ae1d r006-eee5a5d1-e66f-4816-8545-753f172acc6b r006-48e38184-7b51-4c33-aeb4-f380a106e814 r006-010e006f-8b88-4869-9db4-ab0b62fdb860
23:51:30 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: ibmcloud is vols --resource-group-name clacroix-ibm-rv5gk --output json
23:51:31 - MainThread - ocs_ci.utility.retry - WARNING  - Volumes still exist in resource group, Retrying in 10 seconds...
23:51:41 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: ibmcloud is vols --resource-group-name clacroix-ibm-rv5gk --output json
23:51:42 - MainThread - ocs_ci.utility.retry - WARNING  - Volumes still exist in resource group, Retrying in 10 seconds...
23:51:52 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: ibmcloud is vols --resource-group-name clacroix-ibm-rv5gk --output json
23:51:53 - MainThread - ocs_ci.utility.retry - WARNING  - Volumes still exist in resource group, Retrying in 10 seconds...
23:52:03 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: ibmcloud is vols --resource-group-name clacroix-ibm-rv5gk --output json
23:52:04 - MainThread - ocs_ci.deployment.ibmcloud - INFO  - Deleting resource group: clacroix-ibm-rv5gk
23:52:04 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: ibmcloud resource group-delete clacroix-ibm-rv5gk -f
```